src/www/ui/api/Controllers/CopyrightController.php

```diff
@@ -0,0 +1,168 @@
+  {
+    $uploadPk = $args["id"];
+    $uploadTreeId = $args["itemId"];
+    $query = $request->getQueryParams();
+    $limit = $request->getHeaderLine(self::LIMIT_PARAM);
+    $statusVal = true;
+    $returnVal = null;
+    $finalVal = [];
+    if (!empty($limit)) {
+      $limit = filter_var($limit, FILTER_VALIDATE_INT);
+      if ($limit < 1) {
+        $info = new Info(
+          400,
+          "limit should be positive integer > 1",
+          InfoType::ERROR
+        );
+        $limit = self::COPYRIGHT_FETCH_LIMIT;
+      }
+    } else {
+      $limit = self::COPYRIGHT_FETCH_LIMIT;
+    }
+    if (!array_key_exists(self::COPYRIGHT_PARAM, $query)) {
+      $returnVal = new Info(400, "Bad Request. 'status' is a required query param with expected values 'active' or 'inactive", InfoType::ERROR);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+    $status = $query[self::COPYRIGHT_PARAM];
+    if ($status == "active") {
+      $statusVal = true;
+    } else if ($status == "inactive") {
+      $statusVal = false;
+    } else {
+      $returnVal = new Info(400, "Bad Request. Invalid query parameter, expected values 'active' or 'inactive", InfoType::ERROR);
+      return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+    }
+    try {
+      if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $uploadPk)) {
+        $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+      } else if (!$this->dbHelper->doesIdExist($this->restHelper->getUploadDao()->getuploadTreeTableName($uploadPk), "uploadtree_pk", $uploadTreeId)) {
+        $returnVal = new Info(404, "Item does not exist", InfoType::ERROR);
+      }
+      if ($returnVal !== null) {
+        return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+      }
+      $agentId = $this->copyrightHist->getAgentId($uploadPk, 'copyright_ars');
+      $uploadTreeTableName = $this->restHelper->getUploadDao()->getuploadTreeTableName($uploadPk);
+      $page = $request->getHeaderLine(self::PAGE_PARAM);
+      if (empty($page) && $page != "0") {
+        $page = 1;
+      }
+      if (!empty($page) || $page == "0") {
+        $page = filter_var($page, FILTER_VALIDATE_INT);
+        if ($page <= 0) {
+          $info = new Info(
+            400,
+            "page should be positive integer > 0",
+            InfoType::ERROR
+          );
+        }
+        $offset = $limit * ($page - 1);
+        if ($info !== null) {
+          $retVal = $response->withJson($info->getArray(), $info->getCode());
+          return $retVal;
+        }
+        list($rows, $iTotalDisplayRecords, $iTotalRecords)  = $this->copyrightHist->getCopyrights($uploadPk, $uploadTreeId, $uploadTreeTableName, $agentId, 'statement', 'active', $statusVal, $offset, $limit);
+        foreach ($rows as $row) {
+          $row['copyright_count'] = intval($row['copyright_count']);
+          array_push($finalVal, $row);
```

Comment:
Please use more efficient way
```suggestion
          $finalVal[] = $row;
```